test/provisioner/provision_libvirt.go

```diff
@@ -185,3 +214,58 @@ func (l *LibvirtProvisioner) GetStoragePool() (*libvirt.StoragePool, error) {
 
 	return sp, nil
 }
+
+func NewLibvirtInstallOverlay() (*LibvirtInstallOverlay, error) {
+	overlay, err := NewKustomizeOverlay("../../install/overlays/libvirt")
+	if err != nil {
+		return nil, err
+	}
+
+	return &LibvirtInstallOverlay{
+		overlay: overlay,
+	}, nil
+}
+
+func (lio *LibvirtInstallOverlay) Apply(ctx context.Context, cfg *envconf.Config) error {
+	return lio.overlay.Apply(ctx, cfg)
+}
+
+func (lio *LibvirtInstallOverlay) Delete(ctx context.Context, cfg *envconf.Config) error {
+	return lio.overlay.Delete(ctx, cfg)
+}
+
+// Update install/overlays/libvirt/kustomization.yaml
+func (lio *LibvirtInstallOverlay) Edit(ctx context.Context, cfg *envconf.Config, properties map[string]string) error {
+	var err error
+
+	// Mapping the internal properties to ConfigMapGenerator properties and their default values.
+	mapProps := map[string][2]string{
+		"network":     {"default", "LIBVIRT_NET"},
+		"storage":     {"default", "LIBVIRT_POOL"},
+		"pause_image": {"", "PAUSE_IMAGE"},
+		"uri":         {"qemu+ssh://root@192.168.122.1/system?no_verify=1", "LIBVIRT_URI"},
+		"vxlan_port":  {"", "VXLAN_PORT"},
```

Comment:
it seems that if PAUSE_IMAGE / VXLAN_PORT are empty we ignore it in the entrypoint.sh but i think that if possible it would be proffered to not set it at all when it's not set

Reply:
Correct, PAUSE_IMAGE/VXLAN_PORT is kept commented in the kustomize.yaml if it is not set.

For example, I used the following properties file (passed via TEST_E2E_PROVISION_FILE):

```
libvirt_uri="qemu+ssh://wmoschet@192.168.122.1/system?no_verify=1"
libvirt_ssh_key_file="id_rsa"
```
and the diff of install/overlays/libvirt/kustomization.yaml shows that PAUSE_IMAGE/VXLAN_PORT is kept commented:

```
 configMapGenerator:
-- name: peer-pods-cm
-  namespace: confidential-containers-system
-  literals:
+- literals:
   - CLOUD_PROVIDER="libvirt"
-  - LIBVIRT_URI="qemu+ssh://root@192.168.122.1/system?no_verify=1" #set
-  - LIBVIRT_NET="default" # set
-  - LIBVIRT_POOL="default" # set
+  - LIBVIRT_URI="qemu+ssh://wmoschet@192.168.122.1/system?no_verify=1"
+  - LIBVIRT_NET="default"
+  - LIBVIRT_POOL="default"
+  name: peer-pods-cm
+  namespace: confidential-containers-system
   #- PAUSE_IMAGE="" # Uncomment and set if you want to use a specific pause image
   #- VXLAN_PORT="" # Uncomment and set if you want to use a specific vxlan port. Defaults to 4789
```

Is this the behavior that you wanted?

Reply:
yes, thanks